stdgo/reflect/Reflect.hx

```diff
@@ -416,30 +435,111 @@ class _GoType__extension {
 		}
 		return false;
 	}
+
+	public static function comparable(gt:Type):Bool {
+		switch (gt) {
+			case GT_slice(_), GT_func(_, _), GT_map(_, _):
+				return false;
+			case GT_array(elem, _):
+				return elem.comparable();
+			case GT_struct(fields):
+				for (fld in fields)
+					if (!fld.type.comparable())
+						return false;
+				return true;
+			case _:
+				return true;
+		}
+	}
+}
+
+function unserializeType(s:String):Type {
+	var unserializer = new Unserializer(s);
+	var ret = unserializer.unserialize();
+	if (Reflect.isEnumValue(ret)) {
+		return ret;
+	}
+	return GT_invalid;
 }
 
-function ptrTo(t:GoType):GoType {
+function ptrTo(t:Type):Type {
 	return GT_ptr(t);
 }
 
-function typeOf(iface:AnyInterface):GoType {
+var namedTypeMap:Map<GoString, NamedTypeData> = [];
+
+function typeOf(iface:AnyInterface):Type {
+	if (iface == null)
+		return GT_unsafePointer;
 	var tn = iface.typeName();
 	var v = iface.value();
-	if (tn == "*unknown") { // TODO move similar code to Pointer.hx
-		var elem = (v : Dynamic).get();
-		var elemT = typeOfAnyHaxe(elem);
-		if (elemT != GT_invalid)
-			return GT_ptr(elemT);
+	switch (tn) {
+		case "*unknown": // TODO this code may be in the wrong place
+			var elem = (v : Dynamic).get();
+			var elemT = typeOfAnyHaxe(elem);
+			if (elemT != GT_invalid)
+				return GT_ptr(elemT);
+		case _:
 	}
-	var gt = makeUnNamedGoType(tn);
+	var gt = makeTypeFromTypeName(tn);
 	if (gt != GT_invalid)
 		return gt;
 	return typeOfAnyHaxe(v);
 }
 
-function typeOfAnyHaxe(v:Dynamic):GoType {
+function makeTypeFromTypeName(typeName:String):Type {
+	// trace("mNBT:", typeName);
+
+	if (StringTools.startsWith(typeName, "wy21:stdgo.reflect.GT_enum"))
```

Comment:
where does "wy21" come from? Is it seralized data?

Reply:
Yes, it is the start of the Haxe serilazed form of a GT_enum.